src/main.cpp

```diff
@@ -74,25 +70,28 @@ struct bin_checker {
   using data_handle_type = robot_arm*;
 
   reference access(data_handle_type ptr, std::ptrdiff_t offset) const {
-    // We know ptr will be valid asynchronously because we construct it on the stack
-    // of main. In real code we might want to use a shared_ptr or something
-    return std::async([=]{
-      return bin_state{ptr->is_bin_occupied(static_cast<int>(offset)), offset_to_coord(static_cast<int>(offset))};
+    // We know ptr will be valid asynchronously because we construct it on the
+    // stack of main. In real code we might want to use a shared_ptr or
+    // something
+    return std::async([=] {
```

Comment:
bold and daring of you

Reply:
Why so?

Reply:
Its implementation-defined, but I think `std::async` usually launches a new thread, so its sort of doodoo. But there's no read/write sync on the data pointed to by ptr. I would guess your `serial::Serial` interfaces do some of their own protection, calling into the same `offset` could be weird?